### ISSUE TITLE
Use bias epilogue in GPU affine operation if CUDA >= 10.1

### ISSUE THREAD

**Issue by kpu** [marian-nmt/marian-dev#680]
https://github.com/marian-nmt/marian-dev/blob/master/src/graph/node_operators_binary.h#L256-L265 is rather inefficient in bias application: it uses a second GEMM with 1s to add the bias term.  

In CUDA 10.1 there is explicit bias support using https://docs.nvidia.com/cuda/cublas/index.html#cublasLtEpilogue_t `CUBLASLT_EPILOGUE_BIAS = 4` "Apply (broadcasted) bias from the bias vector. Bias vector length must match matrix D rows, and it must be packed (i.e., stride between vector elements is 1). Bias vector is broadcasted to all columns and added before applying the final postprocessing."

I'm not sure if it's there in prior versions.

**Comment by emjotde** [comment]
Oh, good to know, will take a look.

**Comment by emjotde** [comment]
Do we bave similar things for MKL?

**Comment by kpu** [comment]
Regarding MKL, paging @sidkashyap-at-Intel

**Comment by ykim362** [comment]
Regarding the CPU, I recall the current implementation was efficient. I have run several different options to add bias, but this one was fastest for the student models on single core.

**Comment by emjotde** [comment]
I had similar experiences with the GPU, but I did not use anything like proposed here.

**Comment by ykim362** [comment]
@emjotde yes, that makes sense. fbgemm also has a bias epilogue, but it didn't help.

**Comment by XapaJIaMnu** [comment]
This is a new feature in cuda 10.1, which was released long after this code was written

**Comment by frankseide** [comment]
It may help more on a GPU than on a CPU, as GPUs have more cores and are therefore more bottlenecked by memory bandwidth.

**Comment by emjotde** [comment]
@XapaJIaMnu absolutely.

**Comment by rhenry-nv** [comment]
I have added a code path for CUDA >= 11 that will use the cublasLt bias fused op when possible.

FYI - cuda versions before 11 only support this op for int8.
